composables/api/useProductList.ts

```diff
@@ -2,28 +2,33 @@ import { useQuery } from 'vue-query';
 import type { Database } from '~/types/generated-types';
 import type { FullProduct } from '~/types/products';
 
-export const useProductList = (count = Infinity) => {
+export const useProductList = (amount = ref(Infinity)) => {
   const client = useSupabaseClient<Database>();
+  const maxAmount = ref<number | null>(0);
 
   const result = useQuery(
-    ['productsList', count],
+    ['productsList', amount],
     async (): Promise<FullProduct[]> => {
-      const { data, error } = await client
+      const { data, error, count } = await client
         .from('products')
-        .select('*, product_specifications (*)')
-        .limit(count);
+        .select('*, product_specifications (*)', { count: 'exact' })
+        .limit(amount.value);
 
       if (error) {
         console.error(error);
         throw new Error(error.message);
       }
+
+      maxAmount.value = count;
```

Comment:
I used `count` returned from database to conditionally show "see more" button on the `/fleet` page.